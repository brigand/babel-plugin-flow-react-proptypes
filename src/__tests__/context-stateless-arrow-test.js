const babel = require('babel-core');
const content = `
var React = require('react');

type Choices = 'option1' | 'option2';

type FooT = {
    x?: Choices
};

type FooC = {
    lang: string,
};

const Foo = (props: FooT, context: FooC) => {
  <div>{props.x}</div>
};

export default Foo;
`;

it('parses second argument as contextTypes', () => {
  const res = babel.transform(content, {
    babelrc: false,
    presets: ['env', 'stage-1', 'react'],
    plugins: ['syntax-flow', require('../')],
  }).code;
  expect(res).toMatchSnapshot();
});
