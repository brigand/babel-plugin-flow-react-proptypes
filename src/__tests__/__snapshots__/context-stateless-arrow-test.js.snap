// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parses second argument as contextTypes 1`] = `
"'use strict';

Object.defineProperty(exports, \\"__esModule\\", {
    value: true
});

var React = require('react');

var Foo = function Foo(props, context) {
    React.createElement(
        'div',
        null,
        props.x
    );
};

Foo.contextTypes = {
    lang: require('prop-types').string.isRequired
};
Foo.propTypes = {
    x: require('prop-types').oneOf(['option1', 'option2'])
};
exports.default = Foo;"
`;
